tests/view/RoomSettingsTest.java

```diff
@@ -0,0 +1,270 @@
+        new RoomSettingsView(roomSettingsViewModel, roomSettingsController, switchViewController);
+    views.add(roomSettingsView.contentPane, roomSettingsView.viewName);
+
+    JFrame jf = new JFrame();
+    jf.setContentPane(roomSettingsView.contentPane);
+    jf.pack();
+    jf.setVisible(true);
+
+    Room activeRoom = DAOTest.createDummyRoom();
+    roomSettingsViewModel.setActiveRoom(activeRoom);
+    roomSettingsViewModel.setUser(DAOTest.createDummyUser());
+
+    var roomsState = roomsViewModel.getState();
+    var availableRooms = roomsState.getAvailableRooms();
+    availableRooms.add(activeRoom);
+    roomsViewModel.setState(roomsState);
+    roomsViewModel.firePropertyChanged();
+
+    sleep(1000);
+
+    roomSettingsView.getRoomName().setText("Test Room 2");
+    sleep(100);
+    roomSettingsView.getSaveButton().doClick();
+    sleep(100);
+    Assert.assertEquals(1, roomsDataAccessObject.changeRoomNameCalls);
+    Assert.assertEquals("Test Room 2", activeRoom.getName());
+
+    // The user of the tests must click no then yes to pass this test
+    roomSettingsView.getDeleteRoomButton().doClick();
+    sleep(100);
+    Assert.assertEquals(0, roomsDataAccessObject.deleteRoomCalls);
+    Assert.assertEquals(1, availableRooms.size());
+
+    roomSettingsView.getDeleteRoomButton().doClick();
+    sleep(100);
+    Assert.assertEquals(1, roomsDataAccessObject.deleteRoomCalls);
+    Assert.assertEquals(0, availableRooms.size());
+
+    roomSettingsView.getBackButton().doClick();
+    sleep(100);
+    Assert.assertEquals(LoggedInView.viewName, viewManagerModel.getActiveView());
+  }
+
+  @Test
+  public void testRoomSettingsFailures() throws InterruptedException {
+    CardLayout cardLayout = new CardLayout();
+    JPanel views = new JPanel(cardLayout);
+    ViewManagerModel viewManagerModel = new ViewManagerModel();
+
+    SwitchViewController switchViewController = SwitchViewUseCaseFactory.create(viewManagerModel);
+
+    LoginUserDataAccessInterface userDataAccessObject =
+        new LoginUserDataAccessInterface() {
+          @Override
+          public User getUser(String email, String password) {
+            return null;
+          }
+
+          @Override
+          public String getAccessToken(String email, String password) {
+            return null;
+          }
+
+          @Override
+          public DisplayUser getDisplayUser(String email) {
+            return null;
+          }
+        };
+    var roomsDataAccessObject =
+        new RoomsDataAccessInterface() {
+
+          @Override
+          public Room getRoomFromId(
+              User user, LoginUserDataAccessInterface userDAO, String roomId) {
+            return null;
+          }
+
+          @Override
+          public Room addRoom(User user, LoginUserDataAccessInterface userDAO, String roomName) {
+            return null;
+          }
+
+          @Override
+          public void deleteRoom(User user, LoginUserDataAccessInterface userDAO, Room room) {
```

Comment:
Where do you test the confirm delete room popup?

Reply:
https://github.com/uoftm/csc207project/pull/62/files/5d67e163733777185ae23c9ffb47cc2130dd565d#diff-473248cd62fb72eefdc394b747b64e2ed668e9ce990137c4c1cef9126ba06b45R139